### ISSUE TITLE
[Feature] Improve Splash screen

### ISSUE THREAD

**Issue by realitix** [UnderwaterApps/overlap2d#302]
First of all, thank you for your amazing work.

I took a look at source code and I saw that SplashScreen doesn't wait for asset loading.
So splash screen disappears so fast that we don't have time to see it, moreover, just after that, windows is freezed waiting asset loading.
Why not use AssetLoader asynchronously and display the loading state ?

**Comment by azakhary** [comment]
So yeah that's a long story, but let me put it here anyway :)

At some point I set up on a quest to make a splash screen, so that it loads all assets and then opens the editor. I wanted splash screen to look good and modern which meant no "window" borders. That was possible to do with a special JPanel setting. So I did that. Unfortunately when assets were loaded and I tried to maximize that panel and add borders and close/minimize buttons to it, I figured it is impossible, because once JPanel is inited with this settings it cannot be changed. So okay I thought I have 2 JPanels, with 2 lwgl app, but soon enough I figured that is impossible as well, because you can't have 2 lwjgl instances running simultaneously. So I ate that with a sad grim face, and thought okay, I show splash panel with it's lwgl app, then I dispose that and show main panel with it's editor lwjgl app. Unfortunately it was impossible to transfer loaded assets from one context to another, it just was not working. So at this point I just gave up on the splash screen, and was out of options. Just deleted all that trickery, and let splash just run load nothing, close, and then run the editor jpanel and load it's own assets. Which rendered splash screen effectively useless. But it was there just for the "looks of it" because I spend all that time making it look good. At this point there is just one "sane" way to solve it. It's to just use clean java-swing for splash screen, without any libGDX. which is a normal way to do it. Unfortunately (and I tried that too) swing is a piece of shit. And it's pain in the ass to do anything in it. If someone can make a swing app that looks like out splash they will be my hero :)

TL; DR: I tried ton of stuff to make it work and failed, so currently splash is just decorative unless some good guy with swing knowledge and experience and good heart, re-makes it in swing.

The End.

**Comment by realitix** [comment]
I like your story and feel sorry for your splash screen. Maybe the new version of lwjgl would be your solution. From what I saw, version 3 allows several windows. And libGDX should be soon compatible. Meantime, maybe splashscreen in center with the freezed window behind would be better than nothing ? Maybe you ever tried !

**Comment by azakhary** [comment]
That might be better then nothing yes I assume :) It just I am still mad at splash-screen to look at it for a while :D

**Comment by realitix** [comment]
I agree with you it could have been a beautiful splash screen ! Do you want me to close this issue or I let it open to not forgot this sad story ?

**Comment by azakhary** [comment]
Let's keep it open, that's an active issue after all, we have a splash screen that is useless, and that has to be fixed. Besides I write so much so that the current state of splash screen is well explained.

**Comment by realitix** [comment]
@azakhary With lwjgl3 you can know make you splash screen!

**Comment by azakhary** [comment]
That would be so cool actually. I can't sadly work on this right now, but I'll accept any PR.
